frontend/document/credocument.lua

```diff
@@ -1370,82 +1373,75 @@ function CreDocument:setupCallCache()
 
     -- reset full cache
     self._callCacheReset = function()
-        self._call_cache = {}
-        self._call_cache_tags_lru = {}
+        -- "Global" cache, a simple key, value store for *this* document
+        self._global_call_cache = {}
+        -- "Tags" cache, an LRU cache of per-tag simple key, value stores. 10 slots.
+        if self._tag_list_call_cache then
+            self._tag_list_call_cache:clear()
+        else
+            self._tag_list_call_cache = lru.new(10, nil, false)
+        end
+        -- i.e., the only thing that follows any sort of LRU eviction logic is the *list* of tag caches.
+        -- Each individual cache itself is just a simple key, value store (i.e., a hash map).
+        -- Points to said per-tag cache for the current tag.
+        self._tag_call_cache = nil
+        -- Stores the key for said current tag
+        self._current_call_cache_tag = nil
+    end
+    self._callCacheDestroy = function()
+        --- @note: Explicitly destroying the references to the caches is apparently necessary to get their content collected by the GC...
+        ---        c.f., https://github.com/koreader/koreader/pull/7634#discussion_r627820424
```

Comment:
They would really never get collected?
(I guess that sometimes, it would need multiple cycles for this to really happen, possibly because of closure/upvalues here, like `self` being reference in all the functions `self._something() = function() there we use self end ` we create - or something :)

Reply:
Yep, never, not on exit, not on the next document open: not until the state is closed on exit :/.